src/sagas/nanoContract.js

```diff
@@ -263,24 +284,38 @@ export function* requestHistoryNanoContract({ payload }) {
     return;
   }
 
-  if (after == null) {
+  if (before == null && after == null) {
     // it clean the history when starting load from the beginning
+    log.debug('Cleaning previous history to start over.');
     yield put(nanoContractHistoryClean({ ncId }));
   }
 
   try {
-    // fetch from fullnode
-    const { history, next } = yield call(fetchHistory, ncId, count, after, wallet);
+    const req = {
+      wallet,
+      ncId,
+      before,
+      after,
+      count: NANO_CONTRACT_TX_HISTORY_SIZE,
+    };
+    const { history } = yield call(fetchHistory, req);
 
-    if (after != null) {
-      // The first load has always `after` equals null. The first load means to be fast,
-      // but the subsequent ones are all request by user and we want slow down multiple
-      // calls to this effect.
+    if (after != null || before != null) {
+      // We want slow down multiple calls to this effect.
```

Comment:
Why? If you dispatch multiple actions at the same time, they will still execute at the same time, just 1000ms later

Reply:
You right! I removed it.